File: java/src/main/java/com/iteso/nintendo/Snorlax.java
package com.iteso.nintendo;

import com.iteso.nintendo.PokemonCharacter;
import com.iteso.nintendo.Tackle;
import com.iteso.nintendo.Slam;

/**
 * Created by AlfredoRodriguez on 2/24/18.
 */
public class Snorlax extends PokemonCharacter {
    /**
     * Hit Points are the maximum life of pokemon.
     */
    public static final int HIT_POINTS = 150;
    /**
     * Defense multiplier value between 0-1.
     */
    public static final double DEFENSE_MULTIPLIER = 0.4;

    /**
     * Snorlax constructor.
     */
    public Snorlax() {
        Tackle tackle = new Tackle();
        Slam slam = new Slam();
        setType("useless");
        setName("Snorlax");
        setHasEvolution(true);
        setSecondAttack(tackle.performAttack());
        setMainAttack(slam.performAttack());
        setHitPoints(HIT_POINTS);
        setDefenseMultiplier(DEFENSE_MULTIPLIER);
        setMainAttackDamage(slam.attackPower());
        setSecondAttackDamage(tackle.attackPower());
    }

    @Override
    public final String evolve() {
        return null;
    }

    @Override
    public final String defend(final int attack) {
        int damage;

        damage = (int) (attack * getDefenseMultiplier());
        int newHP = getHitPoints() - damage;

        String defendMessage = new String("Defending attack, damage caused is "
                + damage + " new HP is " + newHP);

        setHitPoints(newHP);
        return defendMessage;

    }

    @Override
    public final String secondAttack() {

        String attackMessage = new String("Attacking opponent with "
                + getSecondAttack()
                + " causing a damage of " + getSecondAttackDamage());
        return attackMessage;


    }

    @Override
    public final String mainAttack() {
        String attackMessage = new String("Attacking opponent with "
                + getMainAttack()
                + " causing a damage of " + getMainAttackDamage());
        return attackMessage;

    }

    @Override
    public final void setNewAttack(final int attack, final int attackDamage,
                                   final String newAttack) {
        if (attack == 1) {
            setMainAttack(newAttack);
            setMainAttackDamage(attackDamage);
        } else {
            setSecondAttack(newAttack);
            setSecondAttackDamage(attackDamage);
        }
    }


}

File: python/src/main/python/Snorlax.py
from PokemonCharacter import *
from Slam import *
from Tackle import *


class Snorlax(PokemonCharacter):
    def __init__(self):
        self.HIT_POINTS = 150
        self.DEFENSE_MULTIPLIER = 0.4
        self.slam = Slam()
        self.tackle = Tackle()
        self.setType("useless")
        self.setName("Snorlax")
        self.setHasEvolution(False)
        self.setSecondAttack(self.tackle.performAttack())
        self.setMainAttack(self.slam.performAttack())
        self.setHitPoints(self.HIT_POINTS)
        self.setDefenseMultiplier(self.DEFENSE_MULTIPLIER)
        self.setMainAttackDamage(self.slam.attackPower())
        self.setSecondAttackDamage(self.tackle.attackPower())

    def evolve(self):
        return None

    def defend(self, attack):
        damage = (attack * self.getDefenseMultiplier())
        newHP = self.getHitPoints() - damage
        self.setHitPoints(newHP)
        string = "Defending attack, damage caused is "
        string += str(int(damage))
        string += " new HP is "
        string += str(int(newHP))
        return string

    def executeSecondAttack(self):
        attackMessage = "Attacking opponent with "
        attackMessage += self.getSecondAttack()
        attackMessage += " causing a damage of "
        attackMessage += str(self.getSecondAttackDamage())
        return attackMessage

    def executeMainAttack(self):
        attackMessage = "Attacking opponent with "
        attackMessage += self.getMainAttack()
        attackMessage += " causing a damage of "
        attackMessage += str(self.getMainAttackDamage())
        return attackMessage

    def setNewAttack(self, attack, attackDamage, newAttack):
        if (attack == 1):
            self.setMainAttack(newAttack)
            self.setMainAttackDamage(attackDamage)
        else:
            self.setSecondAttack(newAttack)
            self.setSecondAttackDamage(attackDamage)
